lib/user_search_string.rb

```diff
@@ -0,0 +1,6 @@
+class UserSearchString < String
```

Comment:
> inheritance of a core type defined in C

There's nothing onerous about this code.  Very large portions of Ruby are defined (at least partially) in C.  It's also a very common practice in Ruby to inherit from primitives to create a specialized behavior.

> being thoroughly confused at UserSearchStrings returning Strings

`UserSearchString.new` returns a `UserSearchString`, which `.is_a?(String)`